core/src/test/scala/org/apache/spark/sql/store/CatalogConsistencyTest.scala

```diff
@@ -325,4 +327,24 @@ class CatalogConsistencyTest
     assert(r.length == 5)
 
   }
+
+  test("REMOVE_METASTORE_ENTRY procedure should drop table from catalog") {
```

Comment:
You can also test the failure condition using some test flag which, if set, throws the expected exception from getTableOption(). Then unset the test flag at the end of the test.